packages/dataprovider/src/buildVariables.ts

```diff
@@ -182,20 +194,23 @@ const buildNewInputValue = (
         (t) => t.name === fieldObjectType.name,
       ) as IntrospectionInputObjectType;
 
-      // if it has a set modifier, it is an update array
-      const createModifier = fullFieldObjectType?.inputFields.find(
-        (i) => i.name === "create",
-      );
-      const connectModifier = fullFieldObjectType?.inputFields.find(
-        (i) => i.name === "connect",
-      );
+      const isRelationship = fullFieldObjectType?.inputFields.every((i) => {
+        return Object.keys(ModifiersParams).includes(i.name);
+      });
+
       const setModifier = fullFieldObjectType?.inputFields.find(
-        (i) => i.name === "set",
+        (i) => i.name === ModifiersParams.set,
       );
+
       // is it a relation?
-      if (createModifier && connectModifier) {
+      if (isRelationship) {
```

Comment:
so isn't this much more strict then the previous version where we would only check for create and connect?

Reply:
ah, sorry, did not read your issue carefully

Reply:
I think it will be more precise and flexible at the same time. Let me know if it still unclear the motivation behind this PR